app/controllers/downloads_controller.rb

```diff
@@ -2,6 +2,15 @@ class DownloadsController < ApplicationController
   def show
     #.available_repos
     repo = current_user.repos.find(params[:id])
-    send_data Download.from_s3(repo), :filename => repo.name
+    if repo.password.nil? || password_check
+      #instead of test.jpg we will use repo.name
+      send_data Download.from_s3(repo), :filename => "test.jpg"
```

Comment:
Good catch Laszlo, I agree :filename => repo.name is the most current versions, but other than that and some issues with the password in the repos I fell okay to merge in an effort to keep moving forward.